bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/Activator.java

```diff
@@ -32,4 +54,33 @@ public void stop(BundleContext context) throws Exception {
     public static BundleContext getContext() {
         return Activator.context;
     }
+
+    /**
+     * Returns the {@link ThingTypeRegistry}
+     *
+     * @return {@link ThingTypeRegistry} or null
+     */
+    public static ThingTypeRegistry getThingTypeRegistry() {
+        return Activator.thingTypeRegistry;
+    }
+
+    private ServiceBinder channelTypeServiceBinder;
+
+    private ServiceBinder thingTypeServiceBinder;
+
+    @Override
+    public void start(BundleContext context) throws Exception {
+        Activator.context = context;
+        thingTypeServiceBinder = new ServiceBinder(context, new ThingTypeRegistryBinder());
+        thingTypeServiceBinder.open();
+        channelTypeServiceBinder = new ServiceBinder(context, new ChannelTypeRegistryBinder());
+        channelTypeServiceBinder.open();
+    }
+
+    @Override
+    public void stop(BundleContext context) throws Exception {
+        channelTypeServiceBinder.close();
```

Comment:
>
> So if you really claim so much memory, of course it´s better to release it
> explicitly. But I think this is not dedicated to OSGi specifically. Also in
> Servlets or in any other framework you never know when the framework
> releases the reference to your classes.
>
I programmed C (and C++) for embedded devices several years, so I have done
a lot of memory management manually. Things changed moving to Java again...
Sure, I never know when the framework releases the resources, but I know
when my code does not need the resources anymore (and so I would decrement
the reference counter in "clever" C++ classes).

Thank you for your patient. :)